fendermint/app/src/cmd/run.rs

```diff
@@ -196,36 +197,13 @@ async fn run(settings: Settings) -> anyhow::Result<()> {
             own_subnet_id.clone(),
         );
 
-        if topdown_enabled {
-            if let Some(key) = validator_keypair {
-                let parent_finality_votes = parent_finality_votes.clone();
-
-                tracing::info!("starting the parent finality vote gossip loop...");
-                tokio::spawn(async move {
-                    publish_vote_loop(
-                        parent_finality_votes,
-                        settings.ipc.vote_interval,
-                        settings.ipc.vote_timeout,
-                        key,
-                        own_subnet_id,
-                        client,
-                        |height, block_hash| {
-                            AppVote::ParentFinality(IPCParentFinality { height, block_hash })
-                        },
-                    )
-                    .await
-                });
-            }
-        } else {
-            tracing::info!("parent finality vote gossip disabled");
-        }
-
-        tracing::info!("subscribing to gossip...");
+        info!("subscribing to gossip...");
         let rx = service.subscribe();
-        let parent_finality_votes = parent_finality_votes.clone();
-        tokio::spawn(async move {
-            dispatch_resolver_events(rx, parent_finality_votes, topdown_enabled).await;
-        });
+        let topdown_gossip_rx = IPLDTopdownGossipReceiver { rx };
```

Comment:
Isn't `rx`/`client` suppose to be general purpose, and for the entire IPC hierarchy, but here it's being associated with the topdown process?

Reply:
This function is topdown specific, even in the previous implementation.